f/frizzle/comapeo/comapeo_observations.py

```diff
@@ -92,20 +108,50 @@ def fetch_comapeo_projects(
             if project["project_id"] not in comapeo_project_blocklist
         ]
 
-    if not comapeo_projects:
-        raise ValueError("No valid projects found.")
-
     return comapeo_projects
 
 
+@retry(
+    stop_max_attempt_number=3,
+    wait_exponential_multiplier=500,
+)
+def download_attachment(url, headers, save_path):
+    try:
+        response = requests.get(url, headers=headers)
+
+        content_type = response.headers.get("Content-Type", "")
+        extension = mimetypes.guess_extension(content_type) or ""
+
+        file_name = os.path.basename(url) + extension
+        os.makedirs(os.path.dirname(save_path), exist_ok=True)
+        with open(save_path, "wb") as file:
+            file.write(response.content)
+        logger.info("Download completed.")
+        return file_name
+
+    except Exception as e:
+        logger.error(f"Exception during download: {e}")
+        return None
+
+
+def camel_to_snake(name):
+    return re.sub("([a-z0-9])([A-Z])", r"\1_\2", name).lower()
```

Comment:
To get into the practice I suggest adding a quick unit test for this function.

1. `comapeo_observations_test.py` naming convention will be ignored by `wmill sync`
2. modify `tox.ini` to have a new section `[testenv:comapeo_observations]`

Feel free to do this in another PR; or we just agree to handle it in #9

Reply:
As I wrote in the PR body, I wanted to handle unit tests in a different PR.

Reply:
Apologies :-)